api/src/test/java/org/pmiops/workbench/disk/DiskAdminServiceTest.java

```diff
@@ -0,0 +1,434 @@
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.rawls.model.RawlsWorkspaceACL;
+import org.pmiops.workbench.rawls.model.RawlsWorkspaceAccessEntry;
+import org.pmiops.workbench.utils.TestMockFactory;
+import org.pmiops.workbench.utils.mappers.LeonardoMapperImpl;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+
+@DataJpaTest
+public class DiskAdminServiceTest {
+  private static final Instant NOW = FakeClockConfiguration.NOW.toInstant();
+
+  @MockBean private FireCloudService mockFireCloudService;
+  @MockBean private LeonardoApiClient mockLeonardoApiClient;
+  @MockBean private InitialCreditsService mockInitialCreditsService;
+  @MockBean private MailService mockMailService;
+
+  @Autowired private AccessTierDao accessTierDao;
+  @Autowired private CdrVersionDao cdrVersionDao;
+  @Autowired private UserDao userDao;
+  @Autowired private WorkspaceDao workspaceDao;
+
+  @Autowired private DiskAdminService service;
+
+  private DbUser user1;
+  private DbUser user2;
+  private DbWorkspace workspace;
+  private static final WorkbenchConfig config = WorkbenchConfig.createEmptyConfig();
+
+  @TestConfiguration
+  @MockBean(WorkspaceService.class)
+  @Import({
+    FakeClockConfiguration.class,
+    DiskAdminService.class,
+    LeonardoMapperImpl.class,
+  })
+  static class Configuration {
+    @Bean
+    public WorkbenchConfig workbenchConfig() {
+      return config;
+    }
+  }
+
+  @BeforeEach
+  public void setUp() {
+    user1 = new DbUser();
+    user1.setUsername("alice@fake-research-aou.org");
+    user1.setContactEmail("alice@gmail.com");
+    user1 = userDao.save(user1);
+
+    user2 = new DbUser();
+    user2.setUsername("bob@fake-research-aou.org");
+    user2.setContactEmail("bob@gmail.com");
+    user2 = userDao.save(user2);
+
+    DbCdrVersion cdrVersion = createDefaultCdrVersion();
+    accessTierDao.save(cdrVersion.getAccessTier());
+    cdrVersion = cdrVersionDao.save(cdrVersion);
+
+    workspace =
+        workspaceDao.save(
+            TestMockFactory.createDbWorkspaceStub(
+                    TestMockFactory.createWorkspace("ns1", "Workspace Name 1", "workspacename1"),
+                    0L)
+                .setCdrVersion(cdrVersion)
+                .setGoogleProject("proj1"));
+  }
+
+  @AfterEach
+  public void tearDown() {
+    workspaceDao.deleteAll();
+    cdrVersionDao.deleteAll();
+    userDao.deleteAll();
+  }
+
+  @Test
+  public void testCheckPersistentDisksNoDisks() {
+    stubDisks(Collections.emptyList());
+
+    assertDoesNotThrow(() -> service.checkPersistentDisks());
```

Comment:
the previous (controller) version of many of these was more like "assert endpoint return value is OK"